File: 1_course/week2/4_least_common_multiple/c/main.c
/**
 *
 * C implementation to calculate the least common multiple between two numbers
 *
 * (c) Copyright 2019 Clayton J. Wong ( http://www.claytonjwong.com )
 *
 **/

#include <stdio.h>

typedef size_t Type;

Type gcd( Type a, Type b ){
    for( Type t = 0; b != 0; ){
        t = b;
        b = a % b;
        a = t;
    }
    return a;
}

Type lcm( Type a, Type b ){
    return a * b / gcd( a, b );
}

int main(){
    Type a = 0,
         b = 0;
    scanf( "%ld %ld", &a, &b );
    Type ans = lcm( a, b );
    printf( "%ld\n", ans );
    return 0;
}

File: 1_course/week2/4_least_common_multiple/python/main.py
# python3

##
#
# Python implementation to find the least common multiple between two numbers
#
# (c) Copyright 2019 Clayton J. Wong ( http://www.claytonjwong.com )
#
##

def gcd( a, b ):
    while b != 0:
        t = b
        b = a % b
        a = t
    return a

def lcm( a, b ):
    return a * b // gcd( a, b )

if __name__ == '__main__':
    a, b = map( int, input().split() )
    ans = lcm( a, b )
    print( ans )
